tinygrad/runtime/ops_clang.py

```diff
@@ -1,23 +1,54 @@
-import ctypes, subprocess, pathlib, tempfile
+import platform, struct, ctypes, subprocess, pathlib, tempfile, tinygrad.runtime.autogen.libc as libc
+from mmap import PROT_READ, PROT_WRITE, PROT_EXEC, MAP_ANON, MAP_PRIVATE
 from tinygrad.device import Compiled, Compiler, MallocAllocator
-from tinygrad.helpers import cpu_time_execution, cpu_objdump
+from tinygrad.runtime.support.elf import elf_loader
+from tinygrad.helpers import OSX, getenv, cpu_time_execution, cpu_objdump
 from tinygrad.renderer.cstyle import ClangRenderer
 
+MAP_JIT = 0x0800
+global_handle = ctypes.CDLL(None)
+def getbits(value: int, start: int, end: int): return (value >> start) & ((1 << end-start+1) - 1)
+def patchuint32(image: memoryview, ploc: int, new: int): image[ploc:ploc+4] = struct.pack("<I", struct.unpack("<I", image[ploc:ploc+4])[0] | new)
+
 class ClangCompiler(Compiler):
   def __init__(self, cachekey="compile_clang", args:list[str]|None=None, objdump_tool='objdump'):
-    self.args = ['-march=native'] if args is None else args
+    self.args = ['-shared', '-march=native'] if args is None else args
     self.objdump_tool = objdump_tool
     super().__init__(cachekey)
 
   def compile(self, src:str) -> bytes:
     # TODO: remove file write. sadly clang doesn't like the use of /dev/stdout here
     with tempfile.NamedTemporaryFile(delete=True) as output_file:
-      subprocess.check_output(['clang', '-shared', *self.args, '-O2', '-Wall', '-Werror', '-x', 'c', '-fPIC', '-ffreestanding', '-nostdlib',
+      subprocess.check_output(['clang', *self.args, '-O2', '-Wall', '-Werror', '-x', 'c', '-fPIC', '-ffreestanding', '-nostdlib',
                                '-', '-o', str(output_file.name)], input=src.encode('utf-8'))
       return pathlib.Path(output_file.name).read_bytes()
 
   def disassemble(self, lib:bytes): return cpu_objdump(lib, self.objdump_tool)
 
+class ClangJITCompiler(ClangCompiler):
```

Comment:
I'd just make this it's own class, inherit from Compiler and not ClangCompiler.

Since you aren't using the linker, you should be able to direct clang to stdout and not use a file.